openedx/core/djangoapps/programs/tests/test_utils.py

```diff
@@ -967,6 +972,65 @@ def test_multiple_published_course_runs(self):
 
         self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
 
+    def test_learner_eligibility_for_one_click_purchase_entitlement_products(self):
+        """
+        Learner should be eligible for one click purchase if:
+            - program is eligible for one click purchase
+            - There are courses remaining that have not been purchased
+        """
+        course1 = _create_course(self, self.course_price, course_run_count=2, entitlement_count=1)
+        course2 = _create_course(self, self.course_price, course_run_count=2, entitlement_count=1)
+        expected_skus = set([course1['entitlements'][0]['sku'], course2['entitlements'][0]['sku']])
+        program = ProgramFactory(
+            courses=[course1, course2],
+            is_program_eligible_for_one_click_purchase=True,
+            applicable_seat_types=['verified'],
+        )
+        data = ProgramDataExtender(program, self.user).extend()
+        self.assertTrue(data['is_learner_eligible_for_one_click_purchase'])
+        self.assertEqual(set(data['skus']), expected_skus)
+
+    def test_learner_eligibility_for_one_click_purchase_user_entitlements(self):
+        course1 = _create_course(self, self.course_price, course_run_count=2)
```

Comment:
Doesn't this still need to have an entitlement available?

Reply:
Technically no, because we just check the user's existing entitlements, and don't also check that the product is still available. In reality, there would not exist a user entitlement without a corresponding entitlement product that exists, but it does not need to be created for testing purposes because it will not be accessed.

Reply:
I think it would be better to create the test objects in a way that matches how they would be created in production as closely as possible, as the logic may change in the future and break these tests (maybe we add validations or something to ensure an Entitlement exists for every CourseEntitlement, for instance).

Reply:
There is a CourseEntitlementFactory on my Branch right now in the LMS if that will help @McKenzieW